Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -437,18 +437,31 @@ void Node::invalidate_style(StyleInvalidationReason reason)
 
     set_entire_subtree_needs_style_update(true);
 
-    if (reason == StyleInvalidationReason::NodeInsertBefore || reason == StyleInvalidationReason::NodeRemove) {
-        for (auto* sibling = previous_sibling(); sibling; sibling = sibling->previous_sibling()) {
+    bool is_siblings_style_affected_by_structural_changes = false;
```

Comment:
If >=1 sibling is affected, this code invalidates all siblings. Could it just invalidate the ones that are affected instead?

Reply:
good question, in the beginning I was not sure if it's possible to invalidate only part of siblings this way but it seems ok. I changed it to invalidate only affected ones.